daemon/sm.c

```diff
@@ -322,7 +323,8 @@ MTC_STATIC void
 print_liveset(
     MTC_S32     pri,
     PMTC_S8     log_string,
-    MTC_HOSTMAP hostmap);
+    MTC_HOSTMAP hostmap)
+__attribute__((format(printf, 2, 0)));
```

Comment:
Are you sure? I cannot see the parameters, how can be a formatting string? Maybe it's a bug?

Reply:
Okay, never mind, the 0 says there are no parameter to check against. Maybe a kind of trick can be in place like
```c
static inline const char *
check_print_liveset_format(const char *fmt, ...) __attribute__((format(printf, 1, 2))
{
    return fmt;
}

#define print_liveset(pri, fmt, map) print_liveset(pri, check_print_liveset_format(fmt, "dummy"), map)
```

Reply:
The compiler is still not entirely happy about it, because it says that the 'log_string' is not a string literal, whereas in the other function where I used `format(printf,2,3)` it was able to see that `fmt` is a string literal from the caller and didn't warn about it again.

I'll try the macro trick, that should fix it.